src/foam/u2/Element.js

```diff
@@ -2333,7 +2333,7 @@ foam.CLASS({
           var auth = data.__subContext__.auth;
 
           var propName = this.name.toLowerCase();
-          var clsName  = this.forClass_.substring(this.forClass_.lastIndexOf('.') + 1).toLowerCase();
+          var clsName  = data.cls_.id.substring(data.cls_.id.lastIndexOf('.') + 1).toLowerCase();
```

Comment:
You don't have to do this string manipulation if you already have the cls_, you can just do data.cls_.name;